scipy/optimize/_optimize.py

```diff
@@ -3298,7 +3298,7 @@ def _endprint(x, flag, fval, maxfun, xtol, disp):
 
 
 def brute(func, ranges, args=(), Ns=20, full_output=0, finish=fmin,
-          disp=False, workers=1):
+          disp=False, workers=1, *, integrality=None):
```

Comment:
I think it's a matter of personal/reviewer preference for now. I was asked to do it in a PR a while ago. IIRC, the reason stated was that there are so many parameters, it would be crazy for the user to pass them all in as positional. I think I argued that Python gives the user either option, so who are we to make the choice for them?

I think this is my response to my original argument: as developers, it is not unusual for us to intervene to make it harder to use a function incorrectly. For example, we add input validation. One reason is that it is easier to work with validated inputs in our code than working around unusual inputs. For the user, we think the benefits of adding input validation (preventing the user from making a mistake) outweigh the disadvantages (preventing the user from doing something more conveniently).

I think it's like that. If we make arguments keyword only, we give ourselves more flexibility to change things (e.g. parameter order) in the future, and we make it a little harder to use the function in an unhealthy way.